web/src/lib/components/shared-components/side-bar/side-bar-link.svelte

```diff
@@ -1,14 +1,17 @@
 <script lang="ts">
-  import { resolveRoute } from '$app/paths';
+  import { resolve } from '$app/paths';
   import { page } from '$app/state';
+  import type { RouteId } from '$app/types';
   import Icon from '$lib/components/elements/icon.svelte';
   import { mdiChevronDown, mdiChevronLeft } from '@mdi/js';
   import type { Snippet } from 'svelte';
   import { t } from 'svelte-i18n';
 
+  type UserRouteId<T> = T extends `/(user)/${infer TPart}` ? (TPart extends `${string}[${string}` ? never : T) : never;
```

Comment:
Maybe better to add a comment here? 
```suggestion
  // we need to restrict this type because otherwise the route might need some parameter that currently it's not passed in
  type UserRouteId<T> = T extends `/(user)/${infer TPart}` ? (TPart extends `${string}[${string}` ? never : T) : never;
```